java/client/src/org/openqa/selenium/chrome/ChromeOptions.java

```
@@ -89,9 +76,78 @@ public ChromeOptions() {
+      deleteDuplicates((ArrayList) extensions);
+    }
+  }
+
+  private void mergeExtensionFiles(Capabilities extraCapabilities) {
+    try {
+      Field extensionFilesField = ChromeOptions.class.getDeclaredField("extensionFiles");
+      extensionFilesField.setAccessible(true);
+      List<File> extensionFilesList = (List<File>) extensionFilesField.get(extraCapabilities);
+      for (File path: extensionFilesList) {
+        addExtensions(path);
+      }
+      deleteDuplicates((ArrayList) extensionFiles);
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void mergeBinary(Map options) {
+    if (options.get("binary") != null) {
+      if (binary == null){
+        binary = (String) options.get("binary");
+      } else if (options.get("binary") != binary) {
+        throw new IllegalArgumentException("Binaries can't be merged");
+      }
+    }
+  }
+
+  private void mergeExperimentalOptions(Capabilities extraCapabilities) {
+    try {
+      Field experimentalOptionField = ChromeOptions.class.getDeclaredField("experimentalOptions");
+      experimentalOptionField.setAccessible(true);
+      HashMap<String, Object> experimentalOptionList = (HashMap<String, Object>) experimentalOptionField.get(extraCapabilities);
+      for (Map.Entry<String, Object> experimentalOption : experimentalOptionList.entrySet()) {
+        if (experimentalOptions.get(experimentalOption.getKey()) == null) {
+          experimentalOptions.put(experimentalOption.getKey(), experimentalOption.getValue());
+        } else if (!(experimentalOptions.get(experimentalOption.getKey()).toString()
+            .contentEquals(experimentalOption.getValue().toString()))) {
+          throw new IllegalArgumentException("Experimental options can't be merged");
+        }
+      }
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      e.printStackTrace();
```

Please rethrow rather than swallow